src/components/SearchModal/ManageLists.tsx

```diff
@@ -226,12 +246,24 @@ export function ManageLists({
   setImportList: (list: TokenList) => void
   setListUrl: (url: string) => void
 }) {
+  const { chainId } = useActiveWeb3React()
   const theme = useTheme()
 
   const [listUrlInput, setListUrlInput] = useState<string>('')
 
   const lists = useAllLists()
 
+  const tokenCountByListName = useMemo(() => {
+    const result: any = {}
+    Object.values(lists).forEach(({ current: list }) => {
+      if (!list) {
+        return
+      }
+      result[list.name] = list.tokens.filter((token) => token.chainId === chainId).length
```

Comment:
this counting is expensive, add a method that uses a `WeakMap` so we avoid ever repeating if the `WeakMap` is available